magefiles/magefile.go

```diff
@@ -627,41 +643,102 @@ func SetupSourceBuild() {
 	}
 	dockerPipelineObject := tektonObj.(*tektonapi.Pipeline)
 
-	// Update build-source-image param value to true
-	for i := range dockerPipelineObject.PipelineSpec().Params {
-		if dockerPipelineObject.PipelineSpec().Params[i].Name == "build-source-image" {
-			//fmt.Printf("Current build-source-image param default value: %q\n", dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal)
-			dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal = "true"
+	if source_build {
+		// Update build-source-image param value to true
+		for i := range dockerPipelineObject.PipelineSpec().Params {
+			if dockerPipelineObject.PipelineSpec().Params[i].Name == "build-source-image" {
+				//fmt.Printf("Current build-source-image param default value: %q\n", dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal)
+				dockerPipelineObject.PipelineSpec().Params[i].Default.StringVal = "true"
+			}
+		}
+		// Update the source-build task image reference to SOURCE_BUILD_IMAGE
+		var currentSourceTaskBundle string
+		for i := range dockerPipelineObject.PipelineSpec().Tasks {
+			t := &dockerPipelineObject.PipelineSpec().Tasks[i]
+			params := t.TaskRef.Params
+			var lastBundle *tektonapi.Param
+			sourceTask := false
+			for i, param := range params {
+				if param.Name == "bundle" {
+					lastBundle = &t.TaskRef.Params[i]
+				} else if param.Name == "name" && param.Value.StringVal == "source-build" {
+					sourceTask = true
+					//Change the image in the bundle
```

Comment:
Looks like the comment doesn't belong here

Reply:
Removed.